### PR TITLE
add --exit flag to dev/devicelab/bin/test_runner.dart

### LINKED ISSUE #134070: Consider adding a `--no-retry` flag to the devicelab `test` runner command
Per [dev/devicelab/README.md](https://github.com/flutter/flutter/blob/a7997f606e32bc7e54778df50e003f975b85a536/dev/devicelab/README.md#L63), you can run device lab tests locally using `dev/devicelab/bin/test_runner.dart test -t {NAME_OR_PATH_OF_TEST}`.

Failed tests retry up to two times, for a total of three runs. When I'm modifying a devicelab test and debugging locally, I am usually only interested in running the test once, as failures are more likely to be caused by a change I made rather than a flake.

I'd like to add an option or flag to the `test` command to prevent retries. This would save me dev time and reduce the amount of output I need to sift through when running a devicelab test locally.

### PR DESCRIPTION
**Pull request by andrewkolos**
Resolves #134070 

Adds a flag to the `test_runner.dart test` script that will cause the test runner to exit upon first failure (or, said another way, exit without retrying).

This is in parity with the `--exit` flag of `dev/devicelab/bin/run.dart`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/lib/command/test.dart [hunk 1]
```diff
@@ -24,6 +24,11 @@ class TestCommand extends Command<void> {
           'settings in the test case, and will results in error if no device\n'
           'with given ID/ID prefix is found.',
     );
+    argParser.addFlag(
+      'exit',
+      help: 'Exit on the first test failure. Currently flakes are intentionally (though '
+            'incorrectly) not considered to be failures.',
```
Comment (christopherfujino) [inline dev/devicelab/lib/command/test.dart:30]: why incorrectly?

File: dev/devicelab/lib/command/test.dart [hunk 2]
```diff
@@ -78,6 +83,7 @@ class TestCommand extends Command<void> {
     // Prepend '--' to convert args to options when passed to task
     final List<String> taskArgs = taskArgsRaw.map((String taskArg) => '--$taskArg').toList();
     print(taskArgs);
+    print('@andrewkolos ${argResults!['exit']}');
```
Comment (christopherfujino) [inline dev/devicelab/lib/command/test.dart:86]: delete?

File: dev/devicelab/lib/command/test.dart (hunk 1 above)
Comment (andrewkolos) [reply dev/devicelab/lib/command/test.dart:30]: This is copy-pasted from the `--exit` flag in `run.dart`. This verbiage was introduced in a [commit from @Hixie](https://github.com/andrewkolos/flutter/commit/57f097b532e6afeb08e3256a79ff3009a1210281#diff-cfb3295ba5fb046b811746f29aa08bce2188594d44744c2720e39da5016d5292R72). I interpret this as an opinion that tasks should be considered failing if they fail on their first run, regardless of whether or not they pass after retrying.

Comment (Hixie) [reply dev/devicelab/lib/command/test.dart:30]: flakes are by definition failing tests (that sometimes don't report their failure).

**Comment by Hixie** [comment]
test-exempt: test tooling

File: dev/devicelab/lib/command/test.dart (hunk 1 above)
Comment (christopherfujino) [reply dev/devicelab/lib/command/test.dart:30]: ahh, i see

**Review by christopherfujino — APPROVED** [review]
LGTM. BTW, about the print, what I do is I add an invalid todo comment (like `print('debug'); // TODO`), which I know Linux analyze will fail on (it enforces the `// TODO(ldap)` format), so I don't accidentally merge it. The only downside to this approach is it's annoying in editor if your editor highlights lint errors.

**Pull request merged**